src/journal.rs

```diff
@@ -299,6 +299,14 @@ impl Journal {
         unsafe { ffi::sd_journal_flush_matches(self.j) };
         Ok(self)
     }
+
+    /// Waits until the journal gets changed.
+    /// The maximum time this call sleeps may
+    /// be controlled with the timeout_usec parameter.
+    pub fn wait(&mut self, timeout_usec: u64) -> Result<&mut Journal> {
+        sd_try!(ffi::sd_journal_wait(self.j, timeout_usec));
```

Comment:
`sd_journal_wait()` has some additional semantics on positive return values which is being lost here. What about returning a `ChangeEvent::{Nop|Append|Invalidate}` instead?